mochify/lib/load-config.js

```diff
@@ -10,19 +10,42 @@ async function loadConfig(options) {
 
   const default_config_promise = explorer.search();
 
+  let merged;
   if (options.config) {
     const specified = await explorer.load(options.config);
     const config = Object.assign(specified.config, options);
-    return mergeWithDefault(default_config_promise, config);
+    merged = await mergeWithDefault(default_config_promise, config);
+  } else {
+    merged = await mergeWithDefault(default_config_promise, options);
   }
 
-  return mergeWithDefault(default_config_promise, options);
+  const validation_error = validate(merged);
```

Comment:
Is this the right place to introduce something like this?

Reply:
I'd put the validation function into its own file and move the validate call outside of `loadConfig`.